src/preamble.js

```diff
@@ -667,7 +667,11 @@ function getBinaryPromise(binaryFile) {
   // See https://github.com/github/fetch/pull/92#issuecomment-140665932
   // Cordova or Electron apps are typically loaded from a file:// url.
   // So use fetch if it is available and the url is not a file, otherwise fall back to XHR.
-  if (!wasmBinary && (ENVIRONMENT_IS_WEB || ENVIRONMENT_IS_WORKER)) {
+  if (!wasmBinary
+#if SUPPORT_BASE64_EMBEDDING
+      && !isDataURI(binaryFile)
```

Comment:
I see `getBinaryPromise` now checks for a data URI here, but I was also expecting to see `tryParseAsDataURI` for actually getting the data from a data URI. Where does that happen?

Reply:
When this `if` clauses is not true we call through to `getBinarySync` which knows how to actually handle the Data URI.

See line 699 below.

Reply:
Thanks, but what about line 685? We don't test WebView in our CI, so that might be missed.

Reply:
691 will never be reached if `isDataURI(binaryFile)` because that first `if` statement doesn't end until 695.

Basically we completely bypass all async loading when `isDataURI(binaryFile)` .. which I believe is the correct thing to do.

Reply:
I see, thanks. Makes sense.

I am still a little worried about this change as there could be more code paths other than this. In particular user code might call the lower-level methods which now behave differently. But maybe a changelog entry is good enough for that. lgtm with that.

Reply:
Done.  

Although I think because this only effects `-sSINGLE_FILE` and `--embed-mem-file` builds that this is extremely unlikely.